tests/dryrun_test.go

```diff
@@ -152,16 +151,10 @@ var _ = Describe("[sig-compute]Dry-Run requests", decorators.SigCompute, func()
 		var vm *v1.VirtualMachine
 		resource := "virtualmachines"
 
-		newVM := func() *v1.VirtualMachine {
-			vmiImage := cd.ContainerDiskFor(cd.ContainerDiskCirros)
-			vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(vmiImage, "echo Hi\n")
-			vm := libvmi.NewVirtualMachine(vmi)
-			return vm
-		}
-
 		BeforeEach(func() {
-			vm = newVM()
-
+			vmi := libvmifact.NewCirros()
+			vm = libvmi.NewVirtualMachine(vmi)
+			vm.Namespace = testsuite.GetTestNamespace(vmi)
```

Comment:
In general I think its not effective to set namesapce for an object "manually", its better to spesify the namespace on creation instead, it makes the object state true to reality (as it have no namespace before actually created in the cluster), and reduce potential for unexpected behavior (since its specifying once).

The `vmi` is created with empty namespace, which make the result of `testsuite.GetTestNamespace(vmi)` similar to calling it with `nil` - `testsuite.GetTestNamespace(nil)`.

I think we can simplify it by calling the VMI client create with `testsuite.GetTestNamespace(nil)` instead.
(I dont like the fact we call a function with 'nil', its all over the tests code though)

Reply:
Right. But it failed on e2e, since in this test, we're using some other functions. I added it to make the test running.